packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1551,6 +1562,25 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Stops all running snaps, removes all hooks, closes all connections, and
+   * terminates their workers.
+   *
+   * @param statusEvent - The Snap status event that caused the snap to be
+   * stopped.
+   */
+  public async stopAllSnaps(
+    statusEvent:
+      | SnapStatusEvents.Stop
+      | SnapStatusEvents.Crash = SnapStatusEvents.Stop,
+  ): Promise<void> {
+    const runningSnapIds = this.getAllRunningSnapIds();
+    const promises = runningSnapIds.map(async (id) =>
+      this.stopSnap(id, statusEvent),
+    );
+    await Promise.allSettled(promises);
+  }
```

Comment:
```suggestion
    const snaps = Object.values(this.state.snaps)
      .filter((snap) => this.isRunning(snap.id))
    const promises = snaps.map(async (snap) =>
      this.stopSnap(snap.id, statusEvent),
    );
    await Promise.allSettled(promises);
  }
```

Slightly more performant and less duplication if we just do this I think? 🤔 

Instead of a new utility function